e2e/wdio/core/tests/select.e2e-spec.ts

```diff
@@ -0,0 +1,398 @@
+            click(customControlExample + buttons, 0);
+            waitForElDisplayed(option, 4);
+            const arrLength = getElementArrayLength(option);
+
+            for (let i = 4; arrLength > i; i++) {
+                checkElementStates(option, customControlSelectOptions + i, optionTag, i);
+            }
+        });
+    });
+
+    describe('Extended Options', function() {
+        it('should be able to select the option', () => {
+            const textBefore = getText(extendetOptionsExample + displayedText);
+            click(extendetOptionsExample + buttons, 0);
+            waitForElDisplayed(option, 4);
+            click(options(38));
+            const textAfter = getText(extendetOptionsExample + displayedText);
+            expect(textBefore).not.toEqual(textAfter);
+        });
+
+        it('should have visual check', () => {
+            click(extendetOptionsExample + buttons, 0);
+            waitForElDisplayed(option, 4);
+            const arrLength = getElementArrayLength(option);
+
+            for (let i = 4; arrLength > i; i++) {
+                checkElementStates(option, extendedSelectOptions + i, optionTag, i);
+            }
+        });
+    });
+
+    describe('Mobile Mode', function() {
+        it('should be able to select the option', () => {
+            const textBefore = getText(mobileModeExample + displayedText);
+            click(mobileModeExample + buttons, 0);
+            waitForElDisplayed(option);
+            click(options(44));
+            const textAfter = getText(mobileModeExample + displayedText);
+            expect(textBefore).not.toEqual(textAfter);
+        });
+
+        it('should have visual check', () => {
+            click(mobileModeExample + buttons, 0);
+            waitForElDisplayed(option);
+            const arrLength = getElementArrayLength(option);
+
+            for (let i = 0; arrLength > i; i++) {
+                checkElementStates(option, mobileSelectOptions + i, optionTag, i);
+            }
+        });
+    });
+
+    describe('Max Height', function() {
+        it('should be able to select the option', () => {
+            const textBefore = getText(maxHeightExample + displayedText);
+            click(maxHeightExample + buttons, 0);
+            waitForElDisplayed(option, 4);
+            click(options(47));
+            const textAfter = getText(maxHeightExample + displayedText);
+            expect(textBefore).not.toEqual(textAfter);
+        });
+
+        it('should have visual check', () => {
+            click(maxHeightExample + buttons, 0);
+            waitForElDisplayed(option, 4);
+            const arrLength = getElementArrayLength(option);
+
+            for (let i = 4; arrLength > i; i++) {
+                checkElementStates(option, maxHeightSelectOptions + i, optionTag, i);
+            }
+        });
+    });
+
+    describe('Adding and Removing Options', function() {
+        beforeEach(() => {
+            refreshPage();
+        });
```

Comment:
please add retries. Is this needed? you already have it in the beginning of the test